arch/risc-v/src/common/riscv_vpu.S

```diff
@@ -0,0 +1,156 @@
+ * Public Symbols
+ ****************************************************************************/
+
+    .globl        riscv_vpuconfig
+    .globl        riscv_savevpu
+    .globl        riscv_restorevpu
+    .file         "riscv_vpu.S"
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: riscv_vpuconfig
+ *
+ * Description:
+ *   init vpu
+ *
+ * C Function Prototype:
+ *   void riscv_vpuconfig(void);
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   This function does not return anything explicitly.
+ *
+ ****************************************************************************/
+
+    .type        riscv_vpuconfig, function
+
+riscv_vpuconfig:
+  li           a0, MSTATUS_VS_INIT
+  csrs         CSR_STATUS, a0
+
+  fence.i
+  ret
+
+/****************************************************************************
+ * Name: riscv_savevpu
+ *
+ * Description:
+ *   Given the pointer to a register save area (in A0), save the state of the
+ *   floating point registers.
+ *
+ * C Function Prototype:
+ *   void riscv_savevpu(uintptr_t *regs, uintptr_t *fregs);
+ *
+ * Input Parameters:
+ *   regs  - A pointer to the integer registers that contain the status
+ *   fregs - A pointer to the register save area in which to save the
+ *           floating point registers
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+    .type         riscv_savevpu, function
+
+riscv_savevpu:
+
+  REGLOAD    t0, REG_INT_CTX(a0)
+  li         t1, MSTATUS_VS
+  and        t2, t0, t1
+  li         t1, MSTATUS_VS_DIRTY
+#ifdef CONFIG_ARCH_LAZYVPU
+  bne        t2, t1, 1f
+#else
+  blt        t2, t1, 1f
+#endif
+  li         t1, ~MSTATUS_VS
+  and        t0, t0, t1
+  li         t1, MSTATUS_VS_CLEAN
+  or         t0, t0, t1
+  REGSTORE   t0, REG_INT_CTX(a0)
+
+  riscv_savevpu a1
+
+1:
+  ret
+
+/****************************************************************************
+ * Name: riscv_restorevpu
+ *
+ * Description:
+ *   Given the pointer to a register save area (in A0), restore the state of
+ *   the floating point registers.
+ *
+ * C Function Prototype:
+ *   void riscv_restorevpu(uintptr_t *regs, uintptr_t *fregs);
+ *
+ * Input Parameters:
+ *   regs  - A pointer to the integer registers that contain the status
+ *   fregs - A pointer to the register save area containing the floating
+ *           point registers.
+ *
+ * Returned Value:
+ *   This function does not return anything explicitly.  However, it is
+ *   called from interrupt level assembly logic that assumes that r0 is
+ *   preserved.
+ *
+ ****************************************************************************/
+
+    .type        riscv_restorevpu, function
```

Comment:
It would be good to use same alignment as in other assembly files

Reply:
Yes, the coding style is align with [riscv_fpu.S](https://github.com/apache/nuttx/blob/master/arch/risc-v/src/common/riscv_fpu.S)